runtime/src/iree/base/allocator.h

```diff
@@ -290,6 +290,38 @@ static inline iree_allocator_t iree_allocator_inline_arena(
   return v;
 }
 
+//===----------------------------------------------------------------------===//
+// Optional allocators.
+//===----------------------------------------------------------------------===//
+
+#if IREE_ALLOCATOR_ENABLE_MIALLOC
+
+// Default C allocator controller using malloc/free.
+IREE_API_EXPORT iree_status_t
+iree_allocator_mimalloc_ctl(void* self, iree_allocator_command_t command,
+                            const void* params, void** inout_ptr);
+
+// Allocates using the iree_allocator_malloc and iree_allocator_free methods.
+// These will usually be backed by malloc and free.
+static inline iree_allocator_t iree_allocator_mimalloc(void) {
+  iree_allocator_t v = {NULL, iree_allocator_mimalloc_ctl};
+  return v;
+}
+
+#endif  // IREE_ALLOCATOR_ENABLE_MIALLOC
+
+//===----------------------------------------------------------------------===//
+// Default allocator.
+// While we make limited use of hard-coded allocators in library code, tools
+// bindings, and user code often just wants to defer to a default allocator
+// defined at build time. Such code should use iree_allocator_default()
+// as opposed to a specific allocator.
+//===----------------------------------------------------------------------===//
+
+static inline iree_allocator_t iree_allocator_default(void) {
+  return IREE_ALLOCATOR_DEFAULT();
+}
```

Comment:
I'm skeptical of replacing all uses of `iree_allocator_system()` with `iree_allocator_default()` across the project while keeping the existing name. Seems error prone, especially with out of tree code still using the system allocator. Might be another way to roll this sort of change out...

(Would like @benvanik 's thoughts on this)